elintegrostartapp/grails-app/conf/spring/globech/resourcesScriptBean.groovy

```diff
@@ -945,11 +951,17 @@ beans {
                                     downloadTargetFile(){
                                             var allParams = this.state;
                                             var self = this;
-                                            var fileURL = '/translatorAssistant/downloadTargetFile/'+allParams.projectId+allParams.targetLanguage
-                                            var fileLink = document.createElement('a');
-                                            fileLink.href = fileURL;
-                                            document.body.appendChild(fileLink);
-                                            fileLink.click();
+                                            if(this.\$store.state.vueInitDataframe.loggedIn){
+                                                    var fileURL = '/translatorAssistant/downloadTargetFile/'+allParams.projectId+allParams.targetLanguage
+                                                    var fileLink = document.createElement('a');
+                                                    fileLink.href = fileURL;
+                                                    document.body.appendChild(fileLink);
+                                                    fileLink.click();
+                                            }
+                                           else{
+                                                  self.\$router.push("/login-page/0");
```

Comment:
Let's create a generic function, that will wrap this call, so as a developer we do not need to know about specifics of this call.

excon.gotoPage(self, "login-page")

gotoPage(dataframe, page){
      dataframe.\$router.push("/${page}/0");
}